### PR TITLE
Use camelCase identifiers in the JS API

### LINKED ISSUE #325: Programmatic API options should follow Javascript naming conventions
In the programmatic API documentation it lists hyphenated properties for the options object such as "app-copyright" and "app-version". These are not valid identifiers and should be camelCased instead, so that string identifiers and array-notation don't have to be used to reference them.

Likewise, the version-string object takes properties that are Capitalized, which is unusual for Javascript objects (though doesn't run into the same problem of having to be defined with a string identifier like the hyphenated ones).

### PR DESCRIPTION
**Pull request by j-f1**
- [x] I have read the [section in CONTRIBUTING.md about pull requests](https://github.com/electron-userland/electron-packager/blob/master/CONTRIBUTING.md#filing-pull-requests).
- [x] My changes are appropriately documented.
- [ ] My changes have sufficient test coverage.
  - Does it need any tests?
- [x] The test suite passes.

This fixes #325.

### CODE REVIEW

File: common.js [hunk 1]
```diff
@@ -188,6 +187,34 @@ module.exports = {
...
+    'osx-sign': 'osxSign',
+    'protocol-name': 'protocolName',
+    'version-string': 'versionString'
+  },
+
+  camelCase: function camelCase (properties, warn) {
+    Object.keys(module.exports.kebabProperties).forEach(function (key) {
+      var value = module.exports.kebabProperties[key]
+      if (properties.hasOwnProperty(key)) {
+        warn && warning(`The ${key} parameter is deprecated when used from JS, use ${value} instead`)
```
Comment (malept) [inline common.js]: I would prefer it if this was a full-fledged `if` statement.

Comment (malept) [inline common.js]: Also, it should be mentioned that the old property is going to be removed in the next major version.

**Comment by malept** [comment]
@j-f1 It looks like there's a problem in `win32.js`, take a look at the test output.

Also, what part of the testsuite is not running locally?

**Comment by malept** [comment]
Could you also add a NEWS entry? (An entry under both "Changed" and "Deprecated".)

**Comment by j-f1** [comment]
@malept  Sorry for the delay. I’ve added the changes requested.

The problem doesn’t make sense. Presumably, the files are minified somehow (see [this line](https://travis-ci.org/electron-userland/electron-packager/jobs/199956707#L4329)).

When I try to run the test locally, it downloads Electron, then nothing happens.

File: NEWS.md [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 ## Unreleased
 
+### Changed
+* The JS API now uses camelCase identifiers.
```
Comment (malept) [inline NEWS.md]: > Options previously formatted in kebab-case (i.e., with hyphens) are now available in camelCase, per JavaScript naming standards (#580)

File: NEWS.md [hunk 3]
```diff
@@ -2,6 +2,12 @@
...
+
+### Deprecated
+* Using kebab-case keys for the JS API is deprecated. Please use camelCase.
```
Comment (malept) [inline NEWS.md]: Options currently formatted in kebab-case (i.e., with hyphens) are deprecated in favor of their camelCase variants, per JavaScript naming standards (#580)

File: docs/api.md [hunk 4]
```diff
@@ -51,13 +51,13 @@ An array of functions to be called after Electron has been extracted to a tempor
 
 When `true`, sets both [`arch`](#arch) and [`platform`](#platform) to `all`.
 
-##### `app-copyright`
+##### `appCopyright`
```
Comment (malept) [inline docs/api.md]: For the docs, what I typically do when I'm renaming an option is add an entry for the old name with the type and a deprecated message (for example: [`version-string`](https://github.com/electron-userland/electron-packager/blob/v8.5.1/docs/api.md#version-string))

File: common.js [hunk 5]
```diff
@@ -188,6 +187,36 @@ module.exports = {
...
+    'app-version': 'appVersion',
+    'build-version': 'buildVersion',
+    'app-bundle-id': 'appBundleId',
+    'app-category-type': 'appCategoryType',
+    'extend-info': 'extendInfo',
+    'extra-resource': 'extraResource',
+    'helper-bundle-id': 'helperBundleId',
+    'osx-sign': 'osxSign',
+    'protocol-name': 'protocolName',
+    'version-string': 'versionString'
```
Comment (malept) [inline common.js]: There's no point in camel-casing `version-string`, it's deprecated.

**Comment by malept** [comment]
> The problem doesn’t make sense. Presumably, the files are minified somehow

I think this is an artifact of the code coverage module.

File: win32.js [hunk 6]
```diff
@@ -29,23 +29,23 @@ module.exports = {
...
-      if (opts['build-version']) {
-        rcOpts['file-version'] = opts['build-version']
+      if (opts.buildVersion) {
+        rcOpts['file-version'] = opts.buildVersion
       }
 
-      if (opts['app-copyright']) {
-        rcOpts['version-string'].LegalCopyright = opts['app-copyright']
+      if (opts.appCopyright) {
+        rcOpts.versionString.LegalCopyright = opts.appCopyright
```
Comment (malept) [inline win32.js]: This is incorrect, `rcOpts` is for the `rcedit` module. I take it you did a global search/replace?

File: win32.js [hunk 7]
```diff
@@ -29,23 +29,23 @@ module.exports = {
...
       }
 
-      if (opts.icon || opts.win32metadata || opts['version-string'] || opts['app-copyright'] || opts['app-version'] || opts['build-version']) {
+      if (opts.icon || opts.win32metadata || opts.versionString || opts.appCopyright || opts.appVersion || opts.buildVersion) {
```
Comment (malept) [inline win32.js]: Revert the `opts.versionString`, please.

File: docs/api.md (hunk 4 above)
Comment (j-f1) [inline docs/api.md]: @malept Should I place the deprecated names right underneath the new one, or should I group them at the bottom?

Comment (malept) [inline docs/api.md]: Please keep the options alphabetized, so probably place the deprecated options somewhere above their respective new names.

File: docs/api.md [hunk 8]
```diff
@@ -247,38 +279,58 @@ Valid values are listed in [Apple's documentation](https://developer.apple.com/l
...
+
 *Object* or *`true`*
 
 If present, signs OS X target apps when the host platform is OS X and XCode is installed. When the value is `true`, pass default configuration to the signing module. The configuration values listed below can be customized when the value is an `Object`. See [electron-osx-sign](https://www.npmjs.com/package/electron-osx-sign#opts) for more detailed option descriptions and the defaults.
 - `identity` (*String*): The identity used when signing the package via `codesign`.
 - `entitlements` (*String*): The path to the 'parent' entitlements.
 - `entitlements-inherit` (*String*): The path to the 'child' entitlements.
 
+
+
```
Comment (malept) [inline docs/api.md]: Unnecessary whitespace

File: docs/api.md [hunk 9]
```diff
@@ -116,6 +132,12 @@ but are not limited to:
 - `strictSSL` (*Boolean* - default: `true`): Whether SSL certificates are required to be valid when
   downloading Electron.
 
+
+##### `electron-version`
+
+*String* (**deprecated** and will be removed in a future major version, please use the
+[`electronVersion`](#electronVersion) parameter instead)
+
```
Comment (malept) [inline docs/api.md]: This never existed as a JavaScript API option.
